backend/app/services/chat/config/chat_config.py

```
@@ -380,39 +389,94 @@ def _resolve_shell_type(self, bot_crd: Bot) -> str:
+
+                base_url = settings.BACKEND_INTERNAL_URL.rstrip("/")
+                skill_data["binary_download_url"] = (
+                    f"{base_url}/api/internal/skills/{skill.id}/binary"
+                )
+
+        return skill_data
+
+    def _get_bot_skills(
+        self,
+        bot: Kind,
+        user_preload_skills: list = None,
```

Ruff (0.14.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)